scripts/release-binary.sh

```diff
@@ -89,93 +85,78 @@ if [ -n "${DST}" ]; then
-# k8-dbg is the output directory for x86_64 debug builds (-c dbg).
-BAZEL_OUT="$(bazel info output_path)/k8-dbg/bin"
-
-# Build the debug binary.
-BINARY_NAME="${HOME}/envoy-debug-${SHA}.tar.gz"
-SHA256_NAME="${HOME}/envoy-debug-${SHA}.sha256"
-bazel build ${BAZEL_BUILD_ARGS} -c dbg //src/envoy:envoy_tar
-BAZEL_TARGET="${BAZEL_OUT}/src/envoy/envoy_tar.tar.gz"
-cp -f "${BAZEL_TARGET}" "${BINARY_NAME}"
-sha256sum "${BINARY_NAME}" > "${SHA256_NAME}"
-
-if [ -n "${DST}" ]; then
-  # Copy it to the bucket.
-  echo "Copying ${BINARY_NAME} ${SHA256_NAME} to ${DST}/"
-  gsutil cp "${BINARY_NAME}" "${SHA256_NAME}" "${DST}/"
-fi
-
-# Build the debug package.
-BINARY_NAME="${HOME}/istio-proxy-debug-${SHA}.deb"
-SHA256_NAME="${HOME}/istio-proxy-debug-${SHA}.sha256"
-bazel build ${BAZEL_BUILD_ARGS} -c dbg //tools/deb:istio-proxy
-BAZEL_TARGET="${BAZEL_OUT}/tools/deb/istio-proxy.deb"
-cp -f "${BAZEL_TARGET}" "${BINARY_NAME}"
-exit
-sha256sum "${BINARY_NAME}" > "${SHA256_NAME}"
+# k8-opt is the output directory for x86_64 optimized builds (-c opt, so --config=release-symbol and --config=release).
+# k8-dbg is the output directory for -c dbg builds.
+for config in release release-symbol debug
+do
+  PUSH_DOCKER_IMAGE="true"
+  case $config in
+    "release" )
+      CONFIG_PARAMS="--config=release"
+      BINARY_BASE_NAME="envoy-alpha"
+      PACKAGE_BASE_NAME="istio-proxy"
+      BAZEL_OUT="$(bazel info output_path)/k8-opt/bin"
+      ;;
+    "release-symbol")
+      CONFIG_PARAMS="--config=release-symbol"
+      BINARY_BASE_NAME="envoy-symbol"
+      PACKAGE_BASE_NAME=""
+      BAZEL_OUT="$(bazel info output_path)/k8-opt/bin"
+      ;;
+    "asan")
+      # NOTE: libc++ is dynamically linked in this build.
+      PUSH_DOCKER_IMAGE=""
+      CONFIG_PARAMS="${BAZEL_CONFIG_ASAN} --config=release-symbol"
+      BINARY_BASE_NAME="envoy-asan"
+      PACKAGE_BASE_NAME=""
+      BAZEL_OUT="$(bazel info output_path)/k8-opt/bin"
+      ;;
+    "debug")
+      CONFIG_PARAMS="--config=debug"
+      BINARY_BASE_NAME="envoy-debug"
+      PACKAGE_BASE_NAME="istio-proxy-debug"
+      BAZEL_OUT="$(bazel info output_path)/k8-dbg/bin"
+      ;;
+  esac
 
-if [ -n "${DST}" ]; then
-  # Copy it to the bucket.
-  echo "Copying ${BINARY_NAME} ${SHA256_NAME} to ${DST}/"
-  gsutil cp "${BINARY_NAME}" "${SHA256_NAME}" "${DST}/"
-fi
+  export BUILD_CONFIG=${config}
+
+  echo "Building ${config} proxy"
+  BINARY_NAME="${HOME}/${BINARY_BASE_NAME}-${SHA}.tar.gz"
+  SHA256_NAME="${HOME}/${BINARY_BASE_NAME}-${SHA}.sha256"
+  bazel build ${BAZEL_BUILD_ARGS} ${CONFIG_PARAMS} //src/envoy:envoy_tar
+  BAZEL_TARGET="${BAZEL_OUT}/src/envoy/envoy_tar.tar.gz"
+  cp -f "${BAZEL_TARGET}" "${BINARY_NAME}"
+  sha256sum "${BINARY_NAME}" > "${SHA256_NAME}"
+
+  if [ -n "${DST}" ]; then
+    # Copy it to the bucket.
+    echo "Copying ${BINARY_NAME} ${SHA256_NAME} to ${DST}/"
+    gsutil cp "${BINARY_NAME}" "${SHA256_NAME}" "${DST}/"
+  fi
+
+  if [ -n "${PUSH_DOCKER_IMAGE}"]; then
```

Comment:
This should have separate build & push steps, otherwise it's breaking existing `make test_release` target.

Reply:
There is no need to have a separate test release beyond regular bazel build here. I can guard with DST since that seems to be the pattern.

Reply:
Well, we want to verify that you can build the Docker image in `test_release`.

Reply:
it's just tar on top a fetched layer? docker is just a collection of tar layers.

Reply:
It is, but stuff breaks all the time, and there is no harm in testing it before merging and potentially breaking release.

Reply:
Ok, added build command. We can't really test docker images without running docker, which we don't.